packages/lodestar-beacon-state-transition/test/perf/util.ts

```diff
@@ -0,0 +1,143 @@
+    const numValidators = 114038;
+    const numKeyPairs = 100;
+    const keypairs = interopKeypairs(numKeyPairs);
+    state.validators = (Array.from({length: numValidators}, (_, i) => {
+      return {
+        pubkey: keypairs[i % numKeyPairs].pubkey,
+        withdrawalCredentials: Buffer.alloc(32, i),
+        effectiveBalance: BigInt(31000000000),
+        slashed: false,
+        activationEligibilityEpoch: 0,
+        activationEpoch: 0,
+        exitEpoch: Infinity,
+        withdrawableEpoch: Infinity,
+      };
+    }) as unknown) as List<Validator>;
+    state.balances = Array.from({length: numValidators}, () => BigInt(31217089836)) as List<Gwei>;
+    state.randaoMixes = Array.from({length: config.params.EPOCHS_PER_HISTORICAL_VECTOR}, (_, i) => Buffer.alloc(32, i));
+    // no slashings
+    // no attestations
+    // no justificationBits
+    state.previousJustifiedCheckpoint = {
+      epoch: 23635,
+      root: fromHexString("0x3fe60bf06a57b0956cd1f8181d26649cf8bf79e48bf82f55562e04b33d4785d4"),
+    };
+    state.currentJustifiedCheckpoint = {
+      epoch: 23636,
+      root: fromHexString("0x3ba0913d2fb5e4cbcfb0d39eb15803157c1e769d63b8619285d8fdabbd8181c7"),
+    };
+    state.finalizedCheckpoint = {
+      epoch: 23634,
+      root: fromHexString("0x122b8ff579d0c8f8a8b66326bdfec3f685007d2842f01615a0768870961ccc17"),
+    };
+
+    archivedState = config.types.BeaconState.tree.createValue(state);
+    logger.info("Loaded state", {
+      slot: archivedState.slot,
+      numValidators: archivedState.validators.length,
+    });
+    // cache roots
+    archivedState.hashTreeRoot();
+  }
+  return archivedState.clone();
+}
+
+/**
+ * This is generated from Medalla block 756417
+ */
+export async function generatePerformanceBlock(): Promise<TreeBacked<SignedBeaconBlock>> {
+  if (!signedBlock) {
+    const block = config.types.SignedBeaconBlock.defaultValue();
+    const parentState = await generatePerformanceState();
+    const newState = parentState.clone();
+    newState.slot++;
+    block.message.slot = newState.slot;
+    block.message.proposerIndex = getBeaconProposerIndex(config, newState);
+    block.message.parentRoot = config.types.BeaconBlockHeader.hashTreeRoot(parentState.latestBlockHeader);
+    block.message.stateRoot = fromHexString("0x6c86ca3c4c6688cf189421b8a68bf2dbc91521609965e6f4e207d44347061fee");
+    block.message.body.randaoReveal = fromHexString(
+      "0x8a5d2673c48f22f6ed19462efec35645db490df29eed2f56321dbe4a89b2463b0c902095a7ab74a2dc5b7f67edb1a19507ea3d4361d5af9cb0a524945c91638dfd6568841486813a2c45142659d6d9403f5081febb123a7931edbc248b9d0025"
+    );
+    // eth1Data, graffiti, attestations
+    signedBlock = config.types.SignedBeaconBlock.tree.createValue(block);
+    logger.info("Loaded block at slot", signedBlock.message.slot);
+  }
+  return signedBlock.clone();
+}
+
+const CURVE_ORDER = BigInt("52435875175126190479447740508185965837690552500527637822603658699938581184513");
```

Comment:
this part was copied from lodestar-validator, should we move it to lodestar-util?

Reply:
Yeah :+1: